src/citrine/_serialization/properties.py

```diff
@@ -175,22 +229,21 @@ def _set_elements(self, value: typing.Union[SerializedType, DeserializedType]):
         """
 
 
-def _get_base_class(obj: object, key: str) -> type:
+@lru_cache(maxsize=1024 * 1024)
```

Comment:
This caching drops test execution share from 28 % to 6 %.

Reply:
Why 1024*1024?

Reply:
Arbitrarily large while not very large.  A tighter estimate would be total number of property objects in all the files, but I figured trying to guess at that wasn't worth the effort.

Reply:
It just strikes me as a strange way to specify. If you want the cache size to be about a million, why not just say `maxsize=1e6`?

Reply:
A bias towards base 2, but there's no compelling reason.

Reply:
Benchmarking showed saturation around 128, so I changed both caches down to 1024.  If you have strong opinions, I can change that to `2**10` or `10**3` (both are which are int)